examples/auctions/blind_auction.vy

```diff
@@ -0,0 +1,63 @@
+# Blind Auction
+# Adapted to Vyper from [Solidity by Example](https://github.com/ethereum/solidity/blob/develop/docs/solidity-by-example.rst#blind-auction-1)
+
+# Custom struct for bids
+bid: {
+    blindedBid: bytes32,
+    deposit: wei_value
+}
+
+# Auction parameters
+beneficiary: public(address)
+biddingEnd: public(timestamp)
+revealEnd: public(timestamp)
+
+# Set to true at the end of auction, disallowing any new bids
+ended: public(bool)
+
+# State of bids
+bids: public([address])
+
+# Final auction state
+highestBidder: public(bid[address])
+highestBid: public(wei_value)
+
+
+# Create a blinded auction with `_biddingTime` seconds
+# bidding time and `_revealTime` seconds reveal time
+# on behalf of the beneficiary address `_beneficiary`.
+@public
+def __init__(_beneficiary: address, _biddingTime: timedelta, _revealTime: timedelta):
+    self.beneficiary = _beneficiary
+    self.biddingEnd = block.timestamp + _biddingTime
+    self.reveal_end = self.biddingEnd + _revealTime
+
+
+@public
+@payable
+def bid(_blindedBid: bytes32):
+    # Check if bidding period is over
+    assert block.timestamp < self.biddingEnd
+
+    # Add bid to mapping of all bids
+    # bids[]
+
+    # /// Place a blinded bid with `_blindedBid` = keccak256(value,
+    # /// fake, secret).
+    # /// The sent ether is only refunded if the bid is correctly
+    # /// revealed in the revealing phase. The bid is valid if the
+    # /// ether sent together with the bid is at least "value" and
+    # /// "fake" is not true. Setting "fake" to true and sending
+    # /// not the exact amount are ways to hide the real bid but
+    # /// still make the required deposit. The same address can
+    # /// place multiple bids.
+    # function bid(bytes32 _blindedBid)
+    #     public
+    #     payable
+    #     onlyBefore(biddingEnd)
+    # {
+    #     bids[msg.sender].push(Bid({
+    #         blindedBid: _blindedBid,
+    #         deposit: msg.value
+    #     }));
```

Comment:
It looks like the Solidity example uses a dynamic array to store the bidders, which we do not support. It is often better to track this kind of state with a mapping, and then emit an event tracking the bids as they occur. The bid logs are accessible external to the smart contract for whatever dapp/front end you have, so you can collect and iterate over those if desired (listening to the event stream)

Some guidance on this is that for tracking state that doesn't require ordering, a mapping is often better (and cheaper!) than an array.